src/test/java/com/sggc/controllers/SggcControllerIT.java

```diff
@@ -0,0 +1,220 @@
+        GetCommonGamesRequest sggcRequest = new GetCommonGamesRequest();
+        sggcRequest.setMultiplayerOnly(true);
+        String mockSteamId = "someSteamId";
+        sggcRequest.setSteamIds(Collections.singleton(mockSteamId));
+
+
+        final ApiError error = new ApiError(
+                "Exception",
+                "Request body violates validation rules, check error details for more information.",
+                Collections.singleton("More than one Steam id must be included")
+        );
+        SggcResponse expectedResponse = new SggcResponse(false, error);
+
+        mockMvc.perform(post(SGGC_CONTROLLER_URI)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(mapper.writeValueAsString(sggcRequest)))
+                .andExpect(status().isBadRequest())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.content().json(mapper.writeValueAsString(expectedResponse), false));
+    }
+
+    @Test
+    @DisplayName("Given a request is received with an invalid Vanity URL when validating the Vanity URL then a 400 error will be return with an appropriate response object")
+    public void givenARRequestIsReceivedWithAnInvalidVanityUrlWhenValidatingThenA400ErrorWillBeReturnedWithAnAppropriateResponseObject() throws Exception {
+        GetCommonGamesRequest sggcRequest = new GetCommonGamesRequest();
+        sggcRequest.setMultiplayerOnly(true);
+        String mockVanityUrl = "someVanityUrl!%$%£%$R";
+        String mockSteamId = "someSteamId";
+        sggcRequest.setSteamIds(Set.of(mockVanityUrl,mockSteamId));
+
+        ValidationResult expectedValidationError = new ValidationResult(true, mockVanityUrl, VANITY_URL_NOT_ALPHANUMERIC_ERROR_MESSAGE);
+        when(vanityUrlService.validateSteamIdsAndVanityUrls(Set.of(mockVanityUrl,mockSteamId))).thenReturn(List.of(expectedValidationError));
+
+
+        final ApiError error = new ApiError(
+                "ValidationException",
+                "Request body violates validation rules, check error details for more information.",
+                Collections.singleton(expectedValidationError)
+        );
+        SggcResponse expectedResponse = new SggcResponse(false, error);
+
+        mockMvc.perform(post(SGGC_CONTROLLER_URI)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(mapper.writeValueAsString(sggcRequest)))
+                .andExpect(status().isBadRequest())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.content().json(mapper.writeValueAsString(expectedResponse), false));
+    }
+
+    @Test
+    @DisplayName("Given a request with a valid Vanity URL which does not have a corresponding Steam user id when resolving " +
```

Comment:
"Given a request with a valid Vanity URL. When the Steam API cannot find a  corresponding Steam user ID. Then a 400 error will be returned with an appropriate response object.